wal/src/table_kv_impl/table_unit.rs

```diff
@@ -405,38 +429,60 @@ impl TableUnit {
     fn load_last_sequence<T: TableKv>(
         table_kv: &T,
         scan_ctx: ScanContext,
-        region_id: RegionId,
+        region_id: u64,
         table_id: TableId,
         buckets: &[BucketRef],
+        start_sequence: u64,
     ) -> Result<SequenceNumber> {
         debug!(
             "Load last sequence, buckets{:?}, region id:{}, table id:{}",
             buckets, region_id, table_id
         );
 
         // Starts from the latest bucket, find last sequence of given region id.
+        // It is likely that, table has just been moved to an new shard, so we should
+        // pick `start_sequence - 1`(`start_sequence` equal to flushed_sequence + 1)
+        // as the `last_sequence`.
         for bucket in buckets.iter().rev() {
             let table_name = bucket.wal_shard_table(region_id);
 
-            if let Some(seq) = Self::load_last_sequence_from_table(
+            if let Some(sequence) = Self::load_last_sequence_from_table(
                 table_kv,
                 scan_ctx.clone(),
                 table_name,
                 region_id,
                 table_id,
             )? {
-                return Ok(seq);
+                ensure!(
+                    sequence + 1 >= start_sequence,
+                    LoadLastSequence {
+                        table_id,
+                        msg: format!(
+                            "found last sequence, but next sequence after last 
```

Comment:
It seems indent is wrong. How about

```
Found last sequence, but it's smaller than start sequece, last:{}, start:{}
```

Reply:
in fact, it should be "last sequence + 1 < start sequence".
I found `start sequence` is equal to "flushed sequence + 1"...